### PR TITLE
OPSEXP-2116 Approach change for exsitingConfigMap 

### PR DESCRIPTION
**Pull request by slohe1**

### CODE REVIEW

File: charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml [hunk 1]
```diff
@@ -62,17 +62,23 @@ spec:
...
           {{- include "component-security-context" .Values | indent 8 }}
-          envFrom:
-            - configMapRef:
-                name: {{ .Values.repository.existingConfigMap | default (print (include "alfresco-connector-ms365.fullname" .)) }}
+           {{- $repo_details_cm := .Values.repository.existingConfigMap.name | default (print (include "alfresco-connector-ms365.fullname" .)) }}
           env:
-            {{- range $key, $value := (omit .Values.environment "JAVA_OPTS") }}
-            - name: {{ $key }}
-              value: {{ $value }}
-            {{- end }}
```
Comment (alxgomz) [inline charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml:72]: I don't think you should remove that. You still want to have all other options as per the `.environment` values specified. Only JAVA_OPTS are a special case and treated separately to make sure we include the baseUrl option

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml [hunk 2]
```diff
@@ -62,17 +62,25 @@ spec:
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           {{- include "component-security-context" .Values | indent 8 }}
-          envFrom:
-            - configMapRef:
-                name: {{ .Values.repository.existingConfigMap | default (print (include "alfresco-connector-ms365.fullname" .)) }}
+           {{- $repo_details_cm := .Values.repository.existingConfigMap.name | default (include "alfresco-connector-ms365.fullname" .) }}
+            {{- $repo_url := printf "-Dalfresco.base-url=http://%s:%d" "$REPO_HOST" "$REPO_PORT" }}
```
Comment (gionn) [inline charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml:66]:
```suggestion
            {{- $repo_url := printf "-Dalfresco.base-url=http://%s:%s" "$REPO_HOST" "$REPO_PORT" }}
```

Comment (gionn) [reply charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml:66]: %d is for decimal

**Review by gionn — CHANGES_REQUESTED** [review]

File: charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml [hunk 3]
```diff
@@ -62,17 +62,25 @@ spec:
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           {{- include "component-security-context" .Values | indent 8 }}
-          envFrom:
-            - configMapRef:
-                name: {{ .Values.repository.existingConfigMap | default (print (include "alfresco-connector-ms365.fullname" .)) }}
+           {{- $repo_details_cm := .Values.repository.existingConfigMap.name | default (include "alfresco-connector-ms365.fullname" .) }}
+            {{- $repo_url := printf "-Dalfresco.base-url=http://%s:%s" "$REPO_HOST" "$REPO_PORT" }}
```
Comment (alxgomz) [inline charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml:66]: watch your indentation

File: charts/alfresco-connector-ms365/tests/configmap-connector-ms365_test.yaml [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
-        existingConfigMap: myns/mycm
+        existingConfigMap:
+          name: myns/mycm
     asserts:
       - hasDocuments:
           count: 0
         template: config-connector-ms365.yaml
 
-  - it: should render a default config
+  - it: should render custom repo & CSRF config
```
Comment (alxgomz) [inline charts/alfresco-connector-ms365/tests/configmap-connector-ms365_test.yaml:16]: You're not checking CSRF here. Make sure the name is sensible

**Review by alxgomz — CHANGES_REQUESTED** [review]
Please make sure to force push again to remove the updatecli auto commit. We need to change it to a workflow_dispatch only

File: charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml [hunk 5]
```diff
@@ -62,17 +62,25 @@ spec:
...
+          {{- $repo_url := printf "-Dalfresco.base-url=http://%s:%s" "$REPO_HOST" "$REPO_PORT" }}
           env:
             {{- range $key, $value := (omit .Values.environment "JAVA_OPTS") }}
             - name: {{ $key }}
               value: {{ $value }}
             {{- end }}
             - name: JAVA_OPTS
-              value: >-
-                {{ (printf "%s %s" (.Values.environment.JAVA_OPTS | default "") "-Dalfresco.base-url=$REPO_URL") }}
+              value: {{ printf "%s%s" (.Values.environment.JAVA_OPTS | default "") ($repo_url) }}
```
Comment (alxgomz) [inline charts/alfresco-connector-ms365/templates/deployment-connector-ms365.yaml:73]: without a space you may mess up the java command line

```suggestion
              value: {{ printf "%s %s" (.Values.environment.JAVA_OPTS | default "") ($repo_url) }}
```

**Review by alxgomz — CHANGES_REQUESTED** [review]
You still have  the changes to thee sync charts which we don't want (IIRC the postgres upgrade requires more changes)
Make sure to force push your branch (assuming you don't have this change on your local env) to get rid of this.

File: charts/alfresco-sync-service/Chart.yaml [hunk 6]
```diff
@@ -8,7 +8,7 @@ keywords:
 name: alfresco-sync-service
 sources:
   - https://github.com/Alfresco/acs-deployment
-version: 4.2.0
+version: 4.3.0
```
Comment (alxgomz) [inline charts/alfresco-sync-service/Chart.yaml:11]: please make it 4.2.1. I would consider this a bugfix

**Review by alxgomz — CHANGES_REQUESTED** [review]

**Review by alxgomz — APPROVED** [review]

**Review by gionn — APPROVED** [review]

**Pull request merged**